src/Microsoft.AspNetCore.Mvc.RazorPages/Infrastructure/HandlerParameterDescriptor.cs

```diff
@@ -1,16 +1,13 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
-using System;
 using System.Reflection;
 using Microsoft.AspNetCore.Mvc.Abstractions;
 
 namespace Microsoft.AspNetCore.Mvc.RazorPages.Infrastructure
 {
     public class HandlerParameterDescriptor : ParameterDescriptor
     {
-        public object DefaultValue { get; set; }
```

Comment:
This isn't the right level to handle default values. This should be part of the executor like in MVC.

The general flow is that model binding makes a dictionary, and then we 'splat' the dictionary into an array and the executor will fill in the gaps.